apps/evm/lib/evm/configuration.ex

```diff
@@ -1,13 +1,19 @@
 defprotocol EVM.Configuration do
   @moduledoc """
-  Prototcol for defining hardfork configurations.
+  Protocol for defining hardfork configurations.
   """
 
   @type t :: module()
 
+  # EIP2
   @spec contract_creation_cost(t) :: integer()
   def contract_creation_cost(t)
 
+  # EIP2
+  @spec fail_contract_creation?(t) :: boolean()
+  def fail_contract_creation?(t)
```

Comment:
What do you think about naming this something more specific? `fail_contract_creation?` seems like a thing that could happen for many reasons. What do you think about something like `fail_contract_creation_lack_of_gas?` or something like that? I know at that point it's getting really verbose, but I think in this case it might be better to be verbose over it being confusing. What do you think?

Reply:
yes, good idea